golang/sqlboiler/templates/10_resolver.tpl

```diff
@@ -39,6 +39,58 @@ func (r *Resolver) All{{$modelName}}s(ctx context.Context, args struct {
 	return result, nil
 }
 
+// Search{{$modelName}}s retrieves {{$modelName}}s based on the provided search parameters
+func (r *Resolver) Search{{$modelName}}s(ctx context.Context, args struct {
+	Since    *graphql.ID
+	PageSize int32
+	Search *search{{$modelName}}Args
+}) ({{$modelName}}sCollection, error) {
+	result := {{$modelName}}sCollection{}
+	mods := []qm.QueryMod{qm.Limit(int(args.PageSize))}
+	if args.Since != nil {
+		s := string(*args.Since)
+		i, err := strconv.ParseInt(s, 10, 64)
+		if err != nil {
+			return result, err
+		}
+		mods = append(mods, qm.Offset(int(i)))
+	}
+	if args.Search != nil {
+		v := reflect.ValueOf(args.Search).Elem()
+		for i := 0; i < v.NumField(); i++ {
+			field := v.Type().Field(i)
+			value := v.Field(i)
+			if value.IsNil() || !value.IsValid() {
+				continue
+			}
+
+			// Get column name from tags
+			column, hasColumnName := field.Tag.Lookup("json")
+			// Skip if no DB definition
+			if !hasColumnName {
+				continue
+			}
+
+			operator := "="
+			val := value.Elem().Interface()
+			if dataType := field.Type.String(); dataType == "string" || dataType == "*string" {
+				operator = "LIKE"
+				val = fmt.Sprintf("%%%s%%", val)
+			}
+			mods = append(mods, qm.And(fmt.Sprintf("%s %s ?", column, operator), val))
+		}
+	}
```

Comment:
👍 to adding `input search{ ... }Args` but I'm wondering if we can simply have it as optional argument to `All{{$modelName}}s` instead of adding a new `Search{{$modelName}}s`. they seem pretty similar to me

Reply:
or search can be split into a different PR for discussion, and we can merge this PR first, without search? 🙇

Reply:
Implemented in https://github.com/choonkeat/hellocrud/pull/5 🙇